src/overlays/actors/ovl_En_Onpuman/z_en_onpuman.c

```diff
@@ -27,29 +28,150 @@ const ActorInit En_Onpuman_InitVars = {
+    this->unk_2A0 = NULL;
+    this->actionFunc = func_80B121D8;
+    this->actor.velocity.y = -10.0f;
+    this->actor.terminalVelocity = -10.0f;
+    this->actor.gravity = -5.0f;
+}
+
+void EnOnpuman_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOnpuman* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80B11F44(GlobalContext* globalCtx) {
+    Actor* itemAction = globalCtx->actorCtx.actorLists[ACTORCAT_ITEMACTION].first;
+
+    if (itemAction != NULL) {
+        if (itemAction->id != ACTOR_EN_GAKUFU) {
+            itemAction = itemAction->next;
+        }
+        return itemAction;
+    }
+    return 0;
+}
+
+void func_80B11F78(EnOnpuman* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.ocarinaMode == 4) {
+        this->actionFunc = func_80B121D8;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+    } else if (globalCtx->msgCtx.ocarinaMode == 3) {
+        play_sound(NA_SE_SY_CORRECT_CHIME);
+        globalCtx->msgCtx.ocarinaMode = 4;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+        this->actionFunc = func_80B121D8;
+    }
+}
+
+void func_80B1202C(EnOnpuman* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+
+    if ((Message_GetState(&globalCtx->msgCtx) == 5) && (Message_ShouldAdvance(globalCtx) != 0)) {
+        switch (globalCtx->msgCtx.currentTextId) {
+            case 0x8D4:
+                this->unk_2A4 |= 1;
+                func_80151938(globalCtx, 0x8DA);
+                break;
+
+            case 0x8DA:
+                func_80151938(globalCtx, 0x8D6);
+                if (this->unk_2A0 != NULL) {
+                    this->unk_2A0->home.rot.x = 0x50;
+                }
+                break;
+
+            case 0x8D6:
+                this->actionFunc = func_80B11F78;
+                func_80152434(globalCtx, 0x3A);
+                if (this->unk_2A0 != NULL) {
+                    this->unk_2A0->home.rot.x = 0;
+                }
+                break;
+        }
+    }
+    if ((this->unk_2A4 & 1) != 0) {
+        if (this->actor.cutscene == -1) {
+            this->unk_2A4 &= ~1;
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
+            this->unk_2A4 &= 0xFFFE;
+            ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+void func_80B1217C(EnOnpuman* this, GlobalContext* globalCtx) {
+    if ((Message_GetState(&globalCtx->msgCtx) == 5) && (Message_ShouldAdvance(globalCtx) != 0)) {
+        this->actionFunc = func_80B121D8;
+        func_801477B4(globalCtx);
+    }
+}
+
+void func_80B121D8(EnOnpuman* this, GlobalContext* globalCtx) {
+    s16 absoluteValue;
+
+    if (func_800B8718(&this->actor, &globalCtx->state) != 0) {
```

Comment:
Boolean (see z_actor)
```suggestion
    if (func_800B8718(&this->actor, &globalCtx->state)) {
```